pkg/plugin/conntrack/_cprog/conntrack.c

```diff
@@ -0,0 +1,297 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "vmlinux.h"
+#include "compiler.h"
+#include "bpf_helpers.h"
+#include "conntrack.h"
+
+
+/**
+ * The structure representing an ipv4 5-tuple key in the connection tracking map.
+ */
+struct ct_v4_key {
+    __u32 src_ip;
+    __u32 dst_ip;
+    __u16 src_port;
+    __u16 dst_port;
+    __u8 proto;
+};
+/**
+ * The structure representing a connection in the connection tracking map.
+ */
+struct ct_entry {
+    __u32 lifetime; // lifetime stores the time when the connection should be removed from the map.
```

Comment:
is this a unix timestamp or some internal representation? If the former, we should make a project-wide decision whether we care about https://en.wikipedia.org/wiki/Year_2038_problem
and stick to either 32 or 64 bits everywhere

Reply:
don't we need to use 64 bit time since that's what the kernel uses?

Reply:
`lifetime` counts from VM boot time. We derive the actual wall clock timestamp in userspace. @rbtr worked on this 🙂

Reply:
@rbtr Yea, this is basically fine until the year _**2106** if we use unsigned 32-bit_. The kernel and some projects might want to support practically infinite timestamps - it's a judgement call whether we need to. My vote is not really

Reply:
@nddq The name `lifetime` suggests duration - if this is a timestamp I'd rename to something else, eg. `conn_eviction_time` or similar